internal/cmd/controller/agentmanagement/controllers/manageagent/manageagent_test.go

```diff
@@ -107,6 +118,89 @@ func TestOnClusterChangeAffinity(t *testing.T) {
 	}
 }
 
+func TestNewAgentBundle_SortsAgentTolerations(t *testing.T) {
+	// make sure config is set for newAgentBundle
+	config.Set(config.DefaultConfig())
+
+	checkRegisterAddToScheme(t, appsv1.AddToScheme)
+	checkRegisterAddToScheme(t, networkv1.AddToScheme)
+
+	h := &handler{systemNamespace: "fleet-system"}
+
+	unsorted := []corev1.Toleration{
+		{Key: "b", Value: "2", Operator: corev1.TolerationOpExists, Effect: corev1.TaintEffectNoExecute},
+		{Key: "a", Value: "1", Operator: corev1.TolerationOpEqual, Effect: corev1.TaintEffectNoSchedule},
+		{Key: "a", Value: "1", Operator: corev1.TolerationOpExists, Effect: corev1.TaintEffectNoSchedule},
+		{Key: "a", Value: "0", Operator: corev1.TolerationOpEqual, Effect: corev1.TaintEffectNoExecute},
+	}
+
+	cluster := &fleet.Cluster{ObjectMeta: metav1.ObjectMeta{Name: "c1"}, Spec: fleet.ClusterSpec{AgentTolerations: unsorted}}
+
+	wantUser := []corev1.Toleration{
+		{Key: "a", Value: "0", Operator: corev1.TolerationOpEqual, Effect: corev1.TaintEffectNoExecute},
+		{Key: "a", Value: "1", Operator: corev1.TolerationOpEqual, Effect: corev1.TaintEffectNoSchedule},
+		{Key: "a", Value: "1", Operator: corev1.TolerationOpExists, Effect: corev1.TaintEffectNoSchedule},
+		{Key: "b", Value: "2", Operator: corev1.TolerationOpExists, Effect: corev1.TaintEffectNoExecute},
+	}
+
+	// ensure leader election env is set so NewLeaderElectionOptionsWithPrefix doesn't error
+	t.Setenv("FLEET_AGENT_ELECTION_LEASE_DURATION", "15s")
+	t.Setenv("FLEET_AGENT_ELECTION_RENEW_DEADLINE", "10s")
+	t.Setenv("FLEET_AGENT_ELECTION_RETRY_PERIOD", "2s")
+
+	obj, err := h.newAgentBundle("ns", cluster)
+	if err != nil {
+		t.Fatalf("unexpected error from newAgentBundle: %v", err)
+	}
+
+	b, ok := obj.(*fleet.Bundle)
+	if !ok {
+		t.Fatalf("expected bundle object, got %#v", obj)
+	}
+
+	if len(b.Spec.Resources) == 0 {
+		t.Fatalf("bundle resources empty")
+	}
+
+	content := b.Spec.Resources[0].Content
+	docs := strings.Split(content, "\n---\n")
+
+	var found bool
+	for _, d := range docs {
+		var m map[string]interface{}
+		if err := yaml.Unmarshal([]byte(d), &m); err != nil {
+			continue
+		}
+		if kind, _ := m["kind"].(string); kind == "Deployment" {
+			js, err := yaml.YAMLToJSON([]byte(d))
+			if err != nil {
+				t.Fatalf("failed to convert YAML to JSON: %v", err)
+			}
+			var dep appsv1.Deployment
+			if err := json.Unmarshal(js, &dep); err != nil {
+				t.Fatalf("failed to unmarshal deployment json: %v", err)
+			}
```

Comment:
nit: why can't we directly unmarshal the deployment YAML? Is JSON necessary here?